File: bcode/src/main/java/edu/abani/bcode/service/TenantService.java
package edu.abani.bcode.service;

import edu.abani.bcode.entities.Tenant;
import edu.abani.bcode.repo.TenantRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class TenantService {
    @Autowired
    private TenantRepo repo;

    public void saveTenant(Tenant tenant){
        this.repo.save(tenant);
    }

    public List<Tenant> getTenants(){
        return this.repo.findAll();
    }

}

File: seqgen/src/main/kotlin/edu/abani/seqgen/services/TenantService.kt
package edu.abani.seqgen.services

import edu.abani.seqgen.entities.Tenant
import edu.abani.seqgen.repo.TenantRepo
import org.springframework.stereotype.Component

@Component
class TenantService(val tenantRepo: TenantRepo) {

    fun getAllTenants(): List<Tenant> = tenantRepo.findAll()

    fun getTenantById(id: Long): Tenant = tenantRepo.findById( id ).get()

    fun saveTenant(tenant: Tenant): Tenant = tenantRepo.save( tenant )

    fun deleteTenant(id: Long) = println()

    fun deleteAll() = tenantRepo.deleteAll()

    fun updateTenant(updates: Tenant, id: Long): Tenant {
        val user = tenantRepo.findById(id).orElseThrow()
        user.apply {
            name = updates.name
        }
        return tenantRepo.save(user)
    }

}